server/plugin/api_test.go

```diff
@@ -975,3 +975,62 @@ func TestGetUserChannelsForTeam(t *testing.T) {
 		})
 	}
 }
+
+func TestHandleGetGitRepositories(t *testing.T) {
+	defer monkey.UnpatchAll()
+	p := Plugin{}
+	mockAPI := &plugintest.API{}
+	mockCtrl := gomock.NewController(t)
+	mockedClient := mocks.NewMockClient(mockCtrl)
+	p.API = mockAPI
+	p.Client = mockedClient
+	for _, testCase := range []struct {
+		description           string
+		Organization          string
+		Project               string
+		GetGitRepositoriesErr error
+		statusCode            int
+	}{
+		{
+			description:  "HandleGetGitRepositories: valid",
+			Organization: "mockOrganization",
+			Project:      "mockProject",
+			statusCode:   http.StatusOK,
+		},
+		{
+			description:  "HandleGetGitRepositories: Invalid organization or project name",
+			Organization: "mockOrganization",
+			statusCode:   http.StatusBadRequest,
+		},
+		{
+			description:           "HandleGetGitRepositories: GetGitRepositories returns error",
+			Organization:          "mockOrganization",
+			Project:               "mockProject",
+			GetGitRepositoriesErr: errors.New("failed to get git repository branches"),
+			statusCode:            http.StatusInternalServerError,
+		},
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			mockAPI.On("LogError", mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string"))
```

Comment:
Use the util function here.
Ref- https://github.com/Brightscout/mattermost-plugin-servicenow/blob/60af7b5aacd82462f3cb736e523dc0b54d0e2f94/server/testutils/mock.go#L7